pkg/osvscanner/osvscanner.go

```diff
@@ -357,29 +357,53 @@ func scanDebianDocker(r *output.Reporter, query *osv.BatchedQuery, dockerImageNa
 	return nil
 }
 
-// Filters response according to config, returns number of responses removed
-func filterResponse(r *output.Reporter, query osv.BatchedQuery, resp *osv.BatchedResponse, configManager *config.ConfigManager) int {
+// Filters vulnerabilities and groups according to config, returns filtered vulnerabilities and groups
+func filterVulns(r *output.Reporter, configToUse config.Config, vulns []models.Vulnerability, groups []models.GroupInfo) ([]models.Vulnerability, []models.GroupInfo) {
 	hiddenVulns := map[string]config.IgnoreEntry{}
+	outGroups := []models.GroupInfo{}
+	outVulns := []models.Vulnerability{}
 
-	for i, result := range resp.Results {
-		var filteredVulns []osv.MinimalVulnerability
-		configToUse := configManager.Get(r, query.Queries[i].Source.Path)
-		for _, vuln := range result.Vulns {
-			ignore, ignoreLine := configToUse.ShouldIgnore(vuln.ID)
+	for _, group := range groups {
+		keepIDs := []string{}
```

Comment:
We should document the algorithm in comments here a bit more. IT can be a little hard to follow otherwise.

That said,  we can get rid of this complicated `keepIDs` logic if we just default to ignoring all aliases (without supporting the flag to flip it) right?

Reply:
> That said, we can get rid of this complicated keepIDs logic if we just default to ignoring all aliases (without supporting the flag to flip it) right?

Yeah, the logic is simpler without it

Reply:
> We should document the algorithm in comments here a bit more. IT can be a little hard to follow otherwise.

Added some comments into the changed filter function.